nemo/collections/nlp/modules/common/transformer/transformer_generators.py

```diff
@@ -730,13 +729,52 @@ def as_frozen(self):
         Context manager which temporarily freezes embedding, decoder, and log_softmax modules,
         yields control and finally unfreezes the modules.
         """
+        grad_module_list = {'embeddings': {}, 'decoders': {}, 'log_softmaxes': {}, 'encoders': {}}
```

Comment:
is `as_frozen` equivalent to  `torch.inference_mode`

Reply:
It is, however the nlp collection uses it in a weird way where their tests crash completely if i use inference_mode.
Seems somehow they use beam search during training, that causes complete crash since inference_mode makes tensors impossible to do backward on (without cloning at least). So i could add tensor.clone() all over the place and make it work with inference mode but too much effort and this works more generally.

Reply:
as_frozen() is a partial reimplementation of inference_mode() without making tensors un-backpropable.